chainer/distribution.py

```diff
@@ -0,0 +1,260 @@
+import copy
+
+
+class Distribution(object):
+
+    """Interface of Distribution
+
+    `Distribution` is a bass class to treat probability distributions.
+    When initialization, it takes parameter as input.
+    """
+
+    def _copy_to(self, target):
+        target.__dict__ = copy.copy(self.__dict__)
+        return target
+
+    @property
+    def batch_shape(self):
+        """Returns the shape of a sample.
+
+        Returns:
+            ~chainer.Variable: Output variable representing the shape of a
+            sample.
+
+        """
+        raise NotImplementedError
+
+    def cdf(self, x):
+        """Returns Cumulative Distribution Function for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
```

Comment:
It is clear from the type information that it is a variable, so I think the documentation should be more direct (no need of writing "Input variable representing"). I also think "a random variable" is a little bit misleading because `x` can be a deterministic value. How about writing, e.g., "A data point in the domain of the distribution"?